typescript/radix-sdk/src/modules/populate.ts

```diff
@@ -0,0 +1,580 @@
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk';
+import {
+  ManifestBuilder,
+  Value,
+  ValueKind,
+  address,
+  array,
+  bucket,
+  decimal,
+  enumeration,
+  expression,
+  str,
+  tuple,
+  u8,
+  u32,
+  u64,
+  u128,
+} from '@radixdlt/radix-engine-toolkit';
+import { BigNumber } from 'bignumber.js';
+import { Decimal } from 'decimal.js';
+
+import { assert, strip0x } from '@hyperlane-xyz/utils';
+
+import { bytes } from '../utils.js';
+
+import { RadixQuery } from './query.js';
+
+export class RadixPopulate {
+  protected gateway: GatewayApiClient;
+  protected query: RadixQuery;
+  protected packageAddress: string;
+  protected gasAmount: number;
+
+  constructor(
+    gateway: GatewayApiClient,
+    query: RadixQuery,
+    packageAddress: string,
+    gasAmount: number,
+  ) {
+    this.gateway = gateway;
+    this.query = query;
+    this.packageAddress = packageAddress;
+    this.gasAmount = gasAmount;
+  }
+
+  private createCallFunctionManifest(
+    from_address: string,
+    package_address: string | number,
+    blueprint_name: string,
+    function_name: string,
+    args: Value[],
+  ) {
+    return new ManifestBuilder()
+      .callMethod(from_address, 'lock_fee', [decimal(this.gasAmount)])
+      .callFunction(package_address, blueprint_name, function_name, args)
+      .callMethod(from_address, 'try_deposit_batch_or_refund', [
+        expression('EntireWorktop'),
+        enumeration(0),
+      ])
+      .build();
+  }
+
+  private async createCallMethodManifestWithOwner(
+    from_address: string,
+    contract_address: string,
+    method_name: string,
+    args: Value[],
+  ) {
+    const details =
+      await this.gateway.state.getEntityDetailsVaultAggregated(
+        contract_address,
+      );
+
+    const ownerResource = (details.details as any).role_assignments.owner.rule
+      .access_rule.proof_rule.requirement.resource;
+
+    return new ManifestBuilder()
+      .callMethod(from_address, 'lock_fee', [decimal(this.gasAmount)])
```

Comment:
Considering that i see here multiple methods that have these hardcoded strings, what do you think of creating an enum with all the radix methods that we use so we can avoid magic strings? It would also help having some kind of doc comment or reference to the original doc to know what each one of these does